src/libraries/System.Text.RegularExpressions/src/System/Text/RegularExpressions/Symbolic/Algebras/CharSetSolver.cs

```diff
@@ -40,10 +40,23 @@ public BDD CharConstraint(char c, bool ignoreCase = false, string? culture = nul
             else
             {
                 //individual character BDDs are always fixed
-                return _charPredTable[c] ??= CreateSetFrom(c, 15);
+                if (c < _charPredTable.Length)
+                    return _charPredTable[c] ??= CreateBDDFromChar(c);
+                else
+                    return CreateBDDFromChar(c);
             }
         }
 
+        private BDD CreateBDDFromChar(ushort c)
+        {
+            BDD bdd = BDD.True;
+            for (int k = 0; k < 16; k++)
+            {
+                bdd = (c & (1 << k)) == 0 ? GetOrCreateBDD(k, BDD.False, bdd) : GetOrCreateBDD(k, bdd, BDD.False);
```

Comment:
Are there cheaper ways to build up a BDD?  Maybe the caching involved helps, but it seems like otherwise this is going to incrementally build up the BDD by creating 15 intermediate ones that are then thrown away?

Reply:
There are better ways, I think, but this would involve using e.g. a designated array and non-object base representation with own memory-management over that array.
However this incremental build only happens once per ASCII character, I think it is negligible.